packages/upload-client/src/index.js

```diff
@@ -63,13 +67,16 @@ export async function uploadFile(conf, file, options = {}) {
  * has the capability to perform the action.
  *
  * The issuer needs the `store/add` and `upload/add` delegated capability.
- * @param {import('./types.js').FileLike[]} files File data.
+ * @param {Iterable<FileLike> & { sorted?: boolean }} files  Files that should be in the directory.
+ * To ensure determinism in the IPLD encoding, by default these files MUST be sorted by file.name or this function will return a rejected promise .
+ * To explicitly upload unsorted files in an indeterminate way, pass `files` with `files.sorted === false`.
  * @param {import('./types.js').UploadDirectoryOptions} [options]
  */
 export async function uploadDirectory(conf, files, options = {}) {
+  const entries = 'sorted' in files ? files : requireSortedFiles(files)
```

Comment:
If I pass files with `sorted: true` there is nothing that checks the ordering.

Reply:
This was on purpose. I was trying to make the API basically:
* if you pass it something ordered by name, it works
* if you pass it something unordered by name and with no indication you've sorted it how you want, throw an error once we find it is unordered
* if you pass it something unordered by name and with an explicit indication you've already ordered/sorted it intentionally (in some way but not necessarily by file name), allow the potentially nondeterministic ordering

So my intention was that passing `sorted: true` would be that the caller is indicating "I already sorted this in a way that I want, so don't try to check the ordering or order it on my behalf".
`sorted: false` would also be like "It's not sorted but I'm intentionally marking it as not sorted because I know what I'm doing and want to upload a directory of unsorted things"

But what you've described as what you expected is also reasonable so I will make it work that way

Reply:
ACK makes sence, I think it was just the 3 possible values boolean that got me - null/undefined and true and false. Where true/false actually means the same thing in terms of what _we_ do with the files.